airbyte-integrations/connectors/source-twilio/source_twilio/streams.py

```diff
@@ -0,0 +1,376 @@
+
+    def stream_slices(self, **kwargs) -> Iterable[Optional[Mapping[str, any]]]:
+        stream_instance = self.parent_stream(authenticator=self.authenticator)
+        stream_slices = stream_instance.stream_slices(sync_mode=SyncMode.full_refresh, cursor_field=stream_instance.cursor_field)
+        for stream_slice in stream_slices:
+            for item in stream_instance.read_records(
+                sync_mode=SyncMode.full_refresh, stream_slice=stream_slice, cursor_field=stream_instance.cursor_field
+            ):
+                yield {"sid": item["sid"], "account_sid": item["account_sid"]}
+
+
+class Applications(TwilioNestedStream):
+    """https://www.twilio.com/docs/usage/api/applications#read-multiple-application-resources"""
+
+    parent_stream = Accounts
+
+
+class AvailablePhoneNumberCountries(TwilioNestedStream):
+    """
+    https://www.twilio.com/docs/phone-numbers/api/availablephonenumber-resource#read-a-list-of-countries
+
+    List of available phone number countries, as well as local, mobile and toll free numbers
+    may be different on each request, so could not pass the full refresh tests.
+    """
+
+    parent_stream = Accounts
+    data_field = "countries"
+    subresource_uri_key = "available_phone_numbers"
+    primary_key = None
+
+
+class AvailablePhoneNumbersLocal(TwilioNestedStream):
+    """https://www.twilio.com/docs/phone-numbers/api/availablephonenumberlocal-resource#read-multiple-availablephonenumberlocal-resources"""
+
+    parent_stream = AvailablePhoneNumberCountries
+    data_field = "available_phone_numbers"
+    subresource_uri_key = "local"
+    primary_key = None
+
+
+class AvailablePhoneNumbersMobile(TwilioNestedStream):
+    """https://www.twilio.com/docs/phone-numbers/api/availablephonenumber-mobile-resource#read-multiple-availablephonenumbermobile-resources"""
+
+    parent_stream = AvailablePhoneNumberCountries
+    data_field = "available_phone_numbers"
+    subresource_uri_key = "mobile"
+    primary_key = None
+
+
+class AvailablePhoneNumbersTollFree(TwilioNestedStream):
+    """https://www.twilio.com/docs/phone-numbers/api/availablephonenumber-tollfree-resource#read-multiple-availablephonenumbertollfree-resources"""
+
+    parent_stream = AvailablePhoneNumberCountries
+    data_field = "available_phone_numbers"
+    subresource_uri_key = "toll_free"
+    primary_key = None
+
+
+class IncomingPhoneNumbers(TwilioNestedStream):
+    """https://www.twilio.com/docs/phone-numbers/api/incomingphonenumber-resource#read-multiple-incomingphonenumber-resources"""
+
+    parent_stream = Accounts
+
+
+class Keys(TwilioNestedStream):
+    """https://www.twilio.com/docs/usage/api/keys#read-a-key-resource"""
+
+    parent_stream = Accounts
+
+
+class Calls(TwilioNestedStream, IncrementalTwilioStream):
+    """https://www.twilio.com/docs/voice/api/call-resource#create-a-call-resource"""
+
+    parent_stream = Accounts
+    incremental_filter_field = "EndTime>"
```

Comment:
huh why is there a `>` at the end of this? is this part of Twilio's API?

Reply:
confused since I see a mixture of `incremental_filter_fields` with `>` and without.

Reply:
Agree that is confusing, but it is the only way Twilio filtering works. For instance, https://www.twilio.com/docs/usage/api/usage-record#read-multiple-usagerecord-resources support only filters like `StartDate=YYYY-MM-DD`, but the https://www.twilio.com/docs/voice/api/call-resource#create-a-call-resource like `StartTime>YYYY-MM-DD` (or `StartTime<YYYY-MM-DD`, which is not needed for our purposes).

Reply:
Got it, thanks for the explanation. Can you also leave a comment the first time this appears? Something like `This is how Twilio filtering works. See <link> for more info.'